tokio/tests/io_write_all_buf.rs

```diff
@@ -94,3 +94,50 @@ async fn write_buf_err() {
         Bytes::from_static(b"oworld")
     );
 }
+
+#[tokio::test]
+async fn write_all_buf_vectored() {
+    struct Wr {
+        buf: BytesMut,
+    }
+    impl AsyncWrite for Wr {
+        fn poll_write(
+            self: Pin<&mut Self>,
+            _cx: &mut Context<'_>,
+            _buf: &[u8],
+        ) -> Poll<io::Result<usize>> {
+            // When executing `write_all_buf` with this writer,
+            // `poll_write` is not called.
+            panic!("shouldn't be called")
+        }
+        fn poll_flush(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<io::Result<()>> {
+            Ok(()).into()
+        }
+        fn poll_shutdown(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<io::Result<()>> {
+            Ok(()).into()
+        }
+        fn poll_write_vectored(
+            mut self: Pin<&mut Self>,
+            _cx: &mut Context<'_>,
+            bufs: &[io::IoSlice<'_>],
+        ) -> Poll<Result<usize, io::Error>> {
+            for buf in bufs {
+                self.buf.extend_from_slice(buf);
+            }
+            let n = self.buf.len();
+            Ok(n).into()
+        }
+        fn is_write_vectored(&self) -> bool {
+            // Enable vectored write.
+            true
+        }
+    }
+
+    let mut wr = Wr {
+        buf: BytesMut::with_capacity(64),
+    };
+    let mut buf = Bytes::from_static(b"hello world");
```

Comment:
Maybe we should use a buffer for the test that actually returns multiple chunks when calling `.chunks_vectored()` so that we actually utilize the multiple `IoSlices`?

```suggestion
    let mut buf = Bytes::from_static(b"hello")
        .chain(Bytes::from_static(b" "))
        .chain(Bytes::from_static(b"world"));
```